src/DataFramesMeta.jl

```diff
@@ -404,6 +404,10 @@ function transform(g::GroupedDataFrame; kwargs...)
                     throw("If a function returns a vector, the result " * 
                           "must have the same length as the groups it operates on")
                 end
+                if !(out isa AbstractVector)
+                    throw("return value must not change its kind (single " *
```

Comment:
Actually the message needs to be adapted a bit here, since what matters is whether a vector is returned or not. So maybe this:
> return value must be an `` `AbstractVector` `` for all groups or for none of them.

Also, better check that result is a vector before checking its length, which may not be possible for non-vectors. And `elseif` would be appropriate.